packages/interface/src/components/fullscreen-mode/index.js

```diff
@@ -1,51 +1,42 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useEffect, useRef } from '@wordpress/element';
 
-export class FullscreenMode extends Component {
-	componentDidMount() {
-		this.isSticky = false;
-		this.sync();
+const FullscreenMode = ( { isActive } ) => {
+	const isSticky = useRef( false );
 
+	useEffect( () => {
 		// `is-fullscreen-mode` is set in PHP as a body class by Gutenberg, and this causes
 		// `sticky-menu` to be applied by WordPress and prevents the admin menu being scrolled
 		// even if `is-fullscreen-mode` is then removed. Let's remove `sticky-menu` here as
 		// a consequence of the FullscreenMode setup
 		if ( document.body.classList.contains( 'sticky-menu' ) ) {
-			this.isSticky = true;
+			isSticky.current = true;
```

Comment:
Thanks for the update, so if I'm reading properly, isSticky is only used inside that effect, it means it can be just a local variable right?

Reply:
Yes. I think too.

Reply:
Do you mind making that change?

Reply:
That makes sense. I've integrated a useEffect that does cleanup, so isSticky no longer needs a useRef and can be defined in a let.